src/components/Calculator.js

```diff
@@ -1,58 +1,60 @@
 /* eslint-disable class-methods-use-this */
-import React from 'react';
+import React, { useState } from 'react';
 import calculate from '../logic/calculate';
 
-export class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      total: null,
-      next: null,
-      operation: null,
-    };
-  }
+const Calculator = () => {
+  const [object, setObject] = useState({ total: null, next: null, operation: null });
 
-  onChangeHandler = (event) => {
+  const onChangeHandler = (event) => {
     const getForm = document.getElementById('InputNumber');
-    const obj = calculate(this.state, event.target.name);
-    this.setState(obj);
-    if (obj.next) {
-      getForm.value = obj.next;
-    } else if (obj.total) {
-      getForm.value = obj.total;
-    } else if (!obj.total && !obj.next && !obj.operation) {
+    let { total, next, operation } = calculate(object, event.target.name);
+
+    if (total === undefined) {
+      total = object.total;
+    }
+    if (next === undefined) {
+      next = object.next;
+    }
+    if (operation === undefined) {
+      operation = object.operation;
+    }
+
```

Comment:
- [ ] When an operator is clicked twice, an error is thrown on the browser. I think you can add a condition here to handle this error. It is required to make sure the app behaves as expected while refactoring. :pray: 
![image](https://user-images.githubusercontent.com/64894927/129424093-49e042ea-177e-44c7-a3f6-77955c4e94d5.png)